packages/docusaurus-theme-classic/src/theme/hooks/useTabGroupChoice.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useState, useCallback, useEffect} from 'react';
+
+const TAB_CHOICE_STORAGE_KEY = 'docusaurus.tab-choice';
+
+const useTabGroupChoice = () => {
+  const [tabGroupChoice, setChoice] = useState();
+  const setChoiceSyncWithLocalStorage = useCallback(
+    newChoice => {
+      try {
+        localStorage.setItem(TAB_CHOICE_STORAGE_KEY, newChoice);
```

Comment:
`tab-choice` is must be an array in order to maintain sync for several different tabs. 

See https://deploy-preview-2366--docusaurus-2.netlify.com/docs/markdown-features/ for example.

![image](https://user-images.githubusercontent.com/4408379/76141421-249e0f80-6075-11ea-9e5a-cd6a2e810180.png)


![image](https://user-images.githubusercontent.com/4408379/76141444-58793500-6075-11ea-8358-37e6ee9d435f.png)

Reply:
I think a plain array can work if all the tab choices happen to be disjoint. However, if we have something like `['win', 'mac']` and `['win', 'unix']` and we see `['win']` in the array, there is an ambiguity about which `win` it is referring to.

Should we make it an object keyed by serialized `values` array?

Reply:
Not sure if this is necessary, so regular array should work. We assume that users will use the same tab names for similar tabs (Yarn VS npm eg) throughout the website.